components/TextFields/src/MDCTextInputControllerDefault.m

```diff
@@ -1052,6 +1064,17 @@ + (void)setNormalColorDefault:(UIColor *)normalColorDefault {
       normalColorDefault ? normalColorDefault : MDCTextInputDefaultNormalUnderlineColorDefault();
 }
 
+- (NSString *)placeholderText {
+  return _textInput.placeholder;
+}
+
+- (void)setPlaceholderText:(NSString *)placeholderText {
+  _textInput.placeholder = placeholderText;
```

Comment:
Consider checking equality before updating the placeholder animation constraints since that could kickoff animation.
if (_textInput.placeholder && [_textInput.placeholder isEqualString:placeholderText]) {
  return;
}

Reply:
So this one actually doesn't kick off an animation. It just sets the constants and activation of the constraints. But I'll put that equality thing in there to avoid a non-trivial calculation that's a part of that.